pkg/client/cache/fifo.go

```diff
@@ -241,6 +241,17 @@ func (f *FIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will touch all objects to put them into the processing queue
+func (f *FIFO) Resync() error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	for id := range f.items {
+		f.queue = append(f.queue, id)
+	}
+	f.cond.Broadcast()
```

Comment:
This is how the resync is done in kubernetes.  

The resync will literally walk the items and force the higher order code walk through periodically to ensure the cluster state is up to date.  In many cases it's redundant, but it's done on purpose.  

The purpose is so the controllers continually attempt to validate the state of the universe ~= declared state.